lms/djangoapps/learner_dashboard/tests/test_programs.py

```diff
@@ -0,0 +1,138 @@
+"""
+Tests for viewing the programs enrolled by a learner.
+"""
+import datetime
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.test import override_settings
+import httpretty
+from opaque_keys.edx import locator
+import unittest
+from urlparse import urljoin
+
+from openedx.core.djangoapps.programs.tests.mixins import (
+    ProgramsApiConfigMixin,
+    ProgramsDataMixin)
+from openedx.core.djangoapps.programs.models import ProgramsApiConfig
+from oauth2_provider.tests.factories import ClientFactory
+from provider.constants import CONFIDENTIAL
+from student.models import CourseEnrollment
+from student.tests.factories import UserFactory
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+@override_settings(MKTG_URLS={'ROOT': 'http://edx.org'})
+class TestProgramListing(
+        ModuleStoreTestCase,
+        ProgramsApiConfigMixin,
+        ProgramsDataMixin):
+
+    """
+    Unit tests for getting the list of programs enrolled by a logged in user
+    """
+    PASSWORD = 'test'
+
+    def setUp(self):
+        """
+        Add a student
+        """
+        super(TestProgramListing, self).setUp()
+        ClientFactory(name=ProgramsApiConfig.OAUTH2_CLIENT_NAME, client_type=CONFIDENTIAL)
+        self.student = UserFactory()
+        self.create_programs_config(xseries_ad_enabled=True)
+
+    def _create_course_and_enroll(self, student, org, course, run):
+        """ Creates a course and associated enrollment. """
+        course_location = locator.CourseLocator(org, course, run)
+        course = CourseFactory.create(
+            org=course_location.org,
+            number=course_location.course,
+            run=course_location.run
+        )
+        enrollment = CourseEnrollment.enroll(student, course.id)
+        enrollment.created = datetime.datetime(2000, 12, 31, 0, 0, 0, 0)
+        enrollment.save()
+
+    def _get_program_url(self, marketing_slug):
+        '''
+            Helper function to get the program card url
+        '''
```

Comment:
Docstrings should be delimited with triple double quotes. For more, see https://www.python.org/dev/peps/pep-0257/#specification.